src/test/starlark/testdata/dict.sky

```diff
@@ -0,0 +1,51 @@
+# creation
+
+foo = {'a': 1, 'b': [1, 2]}
+bar = dict(a=1, b=[1, 2])
+baz = dict({'a': 1, 'b': [1, 2]})
+
+assert_eq(foo, bar)
+assert_eq(foo, baz)
+
+# get/setdefault
+
+assert_eq(foo.get('a'), 1)
+assert_eq(bar.get('b'), [1, 2])
+assert_eq(baz.get('c'), None)
+assert_eq(baz.setdefault('c', 15), 15)
+
+# items
+
+assert_eq(foo.items(), [('a', 1), ('b', [1, 2])])
+
+# keys
+
+assert_eq(bar.keys(), ['a', 'b'])
+
+# values
+
+assert_eq(baz.values(), [1, [1, 2], 15])
+
+# pop/popitem
+
+assert_eq(foo.pop('a'), 1)
+assert_eq(bar.popitem(), ('a', 1))
+assert_eq(foo, bar)
+assert_eq(foo.pop('a', 0), 0)
+assert_eq(foo.popitem(), ('b', [1, 2]))
+
+
+# update
```

Comment:
Let's add a case for when the same key is present both in the dict / seq of pairs arg and in the **kwargs. (Testing it in Python 3, it looks like the kwarg is supposed to win.)

Let's also have a case where the same key is present multiple times in the seq of pairs (last occurrence wins).

Reply:
It seems that skylark reports repeated keys as an error, as you can see [here](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/syntax/DictionaryLiteral.java#L76).
So that test won't pass if I added it.
Should I modify skylark's code to not report repeated keys as errors and act the same as python?

Reply:
Let's keep the current behavior in Starlark. I believe it's come up in design discussions before, but even if I'm misremembering this isn't the place to change it. :)

We don't need to test the behavior of `{'a': 1, 'a': 2}` here because that's dictionary literals / parser, not the dict() datatype / constructor. But the `[('a', 1), ('a', 2)]` case is relevant -- let's just have the test assert whatever the current behavior is (I guess that it's allowed and set to 2). Thanks!